components/server/src/bitbucket-server/bitbucket-server-api.ts

```diff
@@ -357,24 +358,36 @@ export class BitbucketServerApi {
         if (isCancelled()) {
             return [];
         }
-        const cap = (query?.cap || 0) > 0 ? query.cap! : 10;
-        let requestsLeft = cap;
-        const limit = `limit=${(query?.limit || 0) > 0 ? query.limit! : 1000}&`;
-        const permission = query.permission ? `permission=${query.permission}&` : "";
-        const runQuery = async (params: string) => {
+
+        const fetchRepos = async (params: { [key: string]: string } = {}) => {
             if (isCancelled()) {
                 return [];
             }
+            // Ensure we only load as many pages as requested
+            let requestsLeft = query?.cap ?? 10;
```

Comment:
it should be outside that we cap all requests, not double of cap.

Reply:
My motiviation for having the cap be scoped to this `fetchRepos` fn is to hide some of the implementation details of what happens when you provide a search string. Otherwise a caller needs to know they should provide `2` as a cap to make sure they get a call for matching against repo `name` and `projectname`. This approach ensures we don't load more than 1 page for each call, which is what we'd want. It also ensures when we try to load all repos that it respects that "cap" as well.

Reply:
We could rename as you suggested?

Reply:
What are you suggesting we rename?

Reply:
Ah, I think you were referring to [this](https://github.com/gitpod-io/gitpod/pull/18566#discussion_r1303403592) - will go ahead with that.